dingo/gw/waveform_generator/waveform_generator.py

```diff
@@ -763,6 +766,603 @@ def generate_TD_waveform(self, parameters_lal: Tuple) -> Dict[str, np.ndarray]:
         pol_dict = {"h_plus": h_plus, "h_cross": h_cross}
         return pol_dict
 
+class NewInterfaceWaveformGenerator:
```

Comment:
In general this class is not easy to review since it duplicates much of the code from `WaveformGenerator`. Could it be made to subclass `WaveformGenerator` and only replace certain methods?

Reply:
Hi Stephen, I committed now a change to make NewInterfaceWaveformGenerator a subclass of WaveformGenerator. This has reduced the amount of duplicated code. More duplication can be avoided in the method generate_hplus_hcross_m, which I think can be defined only once, but this requires some extra modifications, I will work on it  (and on the rest of the comments above) tomorrow.

Reply:
Thanks. Yeah I see there is still some room for reducing duplication. If you don't have time to do it now, let me know.

Reply:
In a next version, reducing more core needs some extra thinking on how to manage certain things, so better to spend more time, for a next version